lib/IRGen/IRGenSIL.cpp

```diff
@@ -2393,6 +2402,31 @@ void IRGenSILFunction::visitTryApplyInst(swift::TryApplyInst *i) {
   visitFullApplySite(i);
 }
 
+static void setCXXFunctionTemplateAddr(IRGenSILFunction &IGF,
```

Comment:
It seems like we need some notion of a "foreign specialization" so this code doesn't have to completely special-case the apply path.